tasks/unit_tests/libs/common/worktree_tests.py

```diff
@@ -1,104 +1,147 @@
 
-#         with agent_context(ctx, 'main'):
-#             branch7 = get_default_branch()
+        if re.match(r'git.*checkout.*', command):
+            on_653x = '6.53.x' in command
 
-#         with agent_context(ctx, '6.53.x'):
-#             branch6 = get_default_branch()
+            self._checkout('6.53.x' if on_653x else 'main')
 
-#         self.assertNotEqual(branch6, branch7)
+            return Result(stdout='checked out')
 
-#     def test_context_no_checkout(self):
-#         ctx = get_ctx()
+        return super().run(command, *args, **kwargs)
 
-#         with agent_context(ctx, '6.53.x'):
-#             branch6 = get_default_branch()
 
-#         with agent_context(ctx, 'main'):
-#             branch7 = get_default_branch()
+class TestWorktree(unittest.TestCase):
+    def setUp(self):
+        # Pull only once
+        self.mock_ctx = WorktreeMockContext()
+        self.mock_ctx.reset()
+        self.patch_workdir = patch('tasks.libs.common.worktree.WORKTREE_DIRECTORY', TEST_WORKTREE_DIR)
+        self.patch_workdir.start()
+        self.patch_agentdir = patch('tasks.libs.common.gomodules.agent_working_directory', lambda: TEST_WORKTREE_DIR)
+        self.patch_agentdir.start()
 
-#         with agent_context(ctx, 'main', skip_checkout=True):
-#             branch_no_checkout = get_default_branch()
+        os.environ['AGENT_WORKTREE_NO_PULL'] = '1'
+        init_env(self.mock_ctx, '6.53.x')
 
-#         self.assertNotEqual(branch6, branch7)
-#         self.assertEqual(branch7, branch_no_checkout)
+    def tearDown(self):
+        self.patch_workdir.stop()
+        self.patch_agentdir.stop()
+        self.mock_ctx.reset()
 
-#     def test_context_no_checkout_error(self):
-#         ctx = get_ctx()
+    def test_context_is_worktree_true(self):
+        with agent_context(self.mock_ctx, '6.53.x'):
+            self.assertTrue(is_worktree())
 
-#         with agent_context(ctx, '6.53.x'):
-#             pass
+    def test_context_is_worktree_false(self):
+        self.assertFalse(is_worktree())
 
-#         def switch_context():
-#             # The current branch is not main
-#             with agent_context(ctx, 'main', skip_checkout=True):
-#                 pass
+    def test_context_nested(self):
+        with agent_context(self.mock_ctx, '6.53.x'):
+            with agent_context(self.mock_ctx, '6.53.x'):
+                self.assertTrue(is_worktree())
+            self.assertTrue(is_worktree())
 
-#         self.assertRaises(AssertionError, switch_context)
+    def test_context_pwd(self):
+        ctx = get_ctx()
+
+        with agent_context(self.mock_ctx, None, skip_checkout=True):
+            pwdnone = ctx.run('pwd').stdout
+
+        with agent_context(self.mock_ctx, '6.53.x'):
+            pwd6 = ctx.run('pwd').stdout
+
+        with agent_context(self.mock_ctx, 'main'):
+            pwdmain = ctx.run('pwd').stdout
+
+        self.assertEqual(pwd6, pwdnone)
+        self.assertEqual(pwd6, pwdmain)
```

Comment:
That assertion seems weird. Should we assert is not equal instead?

Reply:
There is only one worktree which is always used in an `agent_context` context manager (even with `None`) so the directory is always the same